File: timber-lint/src/main/java/timber/lint/TimberIssueRegistry.java
package timber.lint;

import com.android.tools.lint.client.api.IssueRegistry;
import com.android.tools.lint.detector.api.ApiKt;
import com.android.tools.lint.detector.api.Issue;
import java.util.Arrays;
import java.util.List;

public final class TimberIssueRegistry extends IssueRegistry {
  @Override public List<Issue> getIssues() {
    return Arrays.asList(WrongTimberUsageDetector.getIssues());
  }

  @Override public int getApi() {
    return ApiKt.CURRENT_API;
  }
}

File: timber-lint/src/main/java/timber/lint/TimberIssueRegistry.kt
package timber.lint

import com.android.tools.lint.client.api.IssueRegistry
import com.android.tools.lint.client.api.Vendor
import com.android.tools.lint.detector.api.CURRENT_API
import com.android.tools.lint.detector.api.Issue
import com.google.auto.service.AutoService

@Suppress("UnstableApiUsage", "unused")
@AutoService(value = [IssueRegistry::class])
class TimberIssueRegistry : IssueRegistry() {
  override val issues: List<Issue>
    get() = WrongTimberUsageDetector.issues.asList()

  override val api: Int
    get() = CURRENT_API

  /**
   * works with Studio 4.0 or later; see
   * [com.android.tools.lint.detector.api.describeApi]
   */
  override val minApi: Int
    get() = 7

  override val vendor = Vendor(
    vendorName = "JakeWharton/timber",
    identifier = "com.jakewharton.timber:timber:{version}",
    feedbackUrl = "https://github.com/JakeWharton/timber/issues",
  )
}